docs/designs/ps_design.md

```diff
@@ -1,375 +1,191 @@
+**Initialization of model parameters**
 
-```proto
-
-message EmbeddingTableInfo{
-    string name = 1;
-    repeated int64 dim = 2;
-    string initializer = 3;
-}
-
-message Model {
-    int64 version = 1;
-    repeated Tensor variables = 2;
-    repeated EmbeddingTableInfo embedding_table_info = 3;
-}
-```
+After starting, PS does not contain any parameter. For model variables, ElasticDL should initialize them before training process. For embedding vectors, ElasticDL adopts lazy initialization, i.e. initialize them when they are needed in training process.
 
-So the RPC service will be defined as following:
+Since a single PS pod may not have enough memory for big model, workers are responsible for random initializing model variables. After initializing, workers push initialized model variables to corresponding PS pod. Here is the RPC call definition for pushing initialized model.
 
 ```proto
 service PServer{
     rpc push_model(Model) returns (google.protobuf.Empty);
-    rpc pull_variable(PullModelRequest) returns (PullModelResponse);
-    rpc pull_embedding_vector(Tensor) returns (Tensor);
-    rpc push_gradient(PushGradientRequest) returns (PushGradientResponse);
 }
 ```
 
-For the details of RPC service definition, please refer to next section.
+**Pull model parameters**
 
-The interfaces of PServer could be like this:
+Since ElasticDL saves parameters in PS, workers should pull parameters from PS in each iteration of training/evaluation process. 
 
+For model variables, a worker needs to pull model variables from all PS pods before the forward-pass.
 
-```python
-class PServer(elasticdl_pb2_grpc.PServerServicer):
-    def __init__(self, kvstore, grad_queue, opt):
-        self.kvstore = KVStore()
-        self.grad_queue = GradientQueue()
-        self.opt = Optimizer(opt, self.kvstore, self.grad_queue)
+For embedding vectors, ElasticDL should only pull embedding vectors that are used in this iteration. This is because embedding vectors used in each iteration only account for a small proportion of the embedding tables. Only when the `call` function of embedding layer is called do we know which embedding vectors will be used in this function. Thus, embedding layer is responsible for pulling embedding vectors from PS in its `call` function.
 
-    def push_model(self, request, _):
-        pass
+Currently, ElasticDL has already implemented its embedding layer in [`elasticdl.layers.embedding`](../../elasticdl/python/elasticdl/layers/embedding.py) module.
 
-    def push_gradient(self, request, _):
-        pass
-        
-    def pull_variable(self, request, _):
-        pass
+Here are the RPC call definitions for pulling model variables and pulling embedding vectors.
 
-    # embedding param is handled lazily
-    def pull_embedding_vector(self, reques, _):
-        pass
+```proto
+service PServer{
+    rpc pull_variable(PullModelRequest) returns (PullModelResponse);
+    rpc pull_embedding_vector(Tensor) returns (Tensor);
```

Comment:
Tensor -> Where do we define this object? Is it the **Tensor** proto from tensorflow?